account_bank_statement_import/models/account_bank_statement_import.py

```diff
@@ -1,366 +0,0 @@
-
-    @api.model
-    def _get_hide_journal_field(self):
-        """ Return False if the journal_id can't be provided by the parsed
-        file and must be provided by the wizard.
-        See account_bank_statement_import_qif """
-        return True
-
-    journal_id = fields.Many2one(
-        'account.journal', string='Journal',
-        help='Accounting journal related to the bank statement you\'re '
-        'importing. It has be be manually chosen for statement formats which '
-        'doesn\'t allow automatic journal detection (QIF for example).')
-    hide_journal_field = fields.Boolean(
-        string='Hide the journal field in the view',
-        compute='_get_hide_journal_field')
-    data_file = fields.Binary(
-        'Bank Statement File', required=True,
-        help='Get you bank statements in electronic format from your bank '
-        'and select them here.')
-
-    @api.multi
-    def import_file(self):
-        """ Process the file chosen in the wizard, create bank statement(s) and
-        go to reconciliation."""
-        self.ensure_one()
-        data_file = base64.b64decode(self.data_file)
-        statement_ids, notifications = self.with_context(
-            active_id=self.id)._import_file(data_file)
-        # dispatch to reconciliation interface
-        action = self.env.ref(
-            'account.action_bank_reconcile_bank_statements')
-        return {
-            'name': action.name,
-            'tag': action.tag,
-            'context': {
-                'statement_ids': statement_ids,
-                'notifications': notifications
-            },
-            'type': 'ir.actions.client',
-        }
-
-    @api.model
-    def _import_file(self, data_file):
-        """ Create bank statement(s) from file."""
-        # The appropriate implementation module returns the required data
-        statement_ids = []
-        notifications = []
-        parse_result = self._parse_file(data_file)
-        # Check for old version result, with separate currency and account
-        if isinstance(parse_result, tuple) and len(parse_result) == 3:
-            (currency_code, account_number, statements) = parse_result
-            for stmt_vals in statements:
-                stmt_vals['currency_code'] = currency_code
-                stmt_vals['account_number'] = account_number
-        else:
-            statements = parse_result
-        # Check raw data:
-        self._check_parsed_data(statements)
-        # Import all statements:
-        for stmt_vals in statements:
-            (statement_id, new_notifications) = (
-                self._import_statement(stmt_vals))
-            if statement_id:
-                statement_ids.append(statement_id)
-            notifications.append(new_notifications)
-        if len(statement_ids) == 0:
-            raise Warning(_('You have already imported that file.'))
-        return statement_ids, notifications
-
-    @api.model
-    def _import_statement(self, stmt_vals):
```

Comment:
We must preserve an override of the base model since we have a more defensive way to do the import.
* The creation of bank_account_journal is disallowed
* The creation of bank accounts for partners is disallowed.